openhands/runtime/client/runtime.py

```diff
@@ -315,14 +323,22 @@ def close(self, close_client: bool = True):
         containers = self.docker_client.containers.list(all=True)
         for container in containers:
             try:
-                if container.name.startswith(self.container_name_prefix):
+                if container.name == self.container_name:
                     logs = container.logs(tail=1000).decode('utf-8')
                     logger.debug(
                         f'==== Container logs ====\n{logs}\n==== End of container logs ===='
                     )
                     container.remove(force=True)
+
+                # If the app doesn't shut down properly, it can leave runtime containers on the system. This ensures
+                # that all 'openhands-sandbox-' containers are removed as well.
+                if rm_all_containers and container.name.startswith(
```

Comment:
I'd position this whole `if` block above the previous one (at line 326) and the other `if` must then be an `elif ...`, so that the code won't try to remove the container twice.

Reply:
Updated